src/main/java/org/sonarsource/scanner/cli/SystemInfo.java

```diff
@@ -35,8 +44,23 @@ static void print(Logs logger) {
     logger.info(os());
     String scannerOpts = system.getenv("SONAR_SCANNER_OPTS");
     if (scannerOpts != null) {
-      logger.info("SONAR_SCANNER_OPTS=" + scannerOpts);
+      logger.info("SONAR_SCANNER_OPTS=" + redactSensitiveArguments(scannerOpts));
+    }
+  }
+
+  private static String redactSensitiveArguments(String scannerOpts) {
+    String[] arguments = scannerOpts.split("\\s+");
```

Comment:
Good learning, I checked the implementation of `String.split()` to better understand, and indeed it actually uses a `Pattern`, so it makes sense to do it directly.

Also I didn't do it for the other case we have under
`String[] elems = argument.split("=");`
because `String.split()` has an optimization for 1-char regex string.